### PR TITLE
New Data Source: aws_redshiftserverless_workgroup

### PR DESCRIPTION
**Pull request by bradsheppard**
### Description
New Data Source for `aws_redshiftserverless_workgroup`.

### Output from Acceptance Testing

```
$ make testacc TESTS=TestAccRedshiftServerlessWorkgroupDataSource PKG=redshiftserverless
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/redshiftserverless/... -v -count 1 -parallel 20 -run='TestAccRedshiftServerlessWorkgroupDataSource'  -timeout 180m
=== RUN   TestAccRedshiftServerlessWorkgroupDataSource_basic
=== PAUSE TestAccRedshiftServerlessWorkgroupDataSource_basic
=== CONT  TestAccRedshiftServerlessWorkgroupDataSource_basic
--- PASS: TestAccRedshiftServerlessWorkgroupDataSource_basic (552.32s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/redshiftserverless	552.406s

...
```

### CODE REVIEW

File: internal/service/redshiftserverless/workgroup_data_source.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"enhanced_vpc_routing": {
+				Type:     schema.TypeBool,
+				Optional: true,
```
Comment (DrFaust92) [inline internal/service/redshiftserverless/workgroup_data_source.go:79]: should be computed

**Review by DrFaust92 — CHANGES_REQUESTED** [review]
Minor issue, otherwise LGTM

**Comment by bradsheppard** [comment]
@DrFaust92 Comments addressed! Also fixed merged conflicts, and switched to the `SDKDataSource` annotation, based on recent changes.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccRedshiftServerlessWorkgroupDataSource_' PKG=redshiftserverless
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/redshiftserverless/... -v -count 1 -parallel 20  -run=TestAccRedshiftServerlessWorkgroupDataSource_ -timeout 180m
=== RUN   TestAccRedshiftServerlessWorkgroupDataSource_basic
=== PAUSE TestAccRedshiftServerlessWorkgroupDataSource_basic
=== CONT  TestAccRedshiftServerlessWorkgroupDataSource_basic
--- PASS: TestAccRedshiftServerlessWorkgroupDataSource_basic (728.21s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/redshiftserverless	733.574s
```

**Pull request merged**